OutOfSchool/OutOfSchool.WebApi/Controllers/V2/WorkshopController.cs

```diff
@@ -200,13 +200,25 @@ public async Task<IActionResult> Create([FromForm] WorkshopV2Dto dto)
     [Consumes("multipart/form-data")]
     public async Task<IActionResult> Update([FromForm] WorkshopV2Dto dto)
     {
+        if (dto == null)
+        {
+            return BadRequest("Workshop is null.");
+        }
+
         var userHasRights = await IsUserProvidersOwner(dto.ProviderId).ConfigureAwait(false);
         if (!userHasRights)
         {
             return StatusCode(StatusCodes.Status403Forbidden, "Forbidden to update workshops for another providers.");
         }
 
-        if (!await combinedWorkshopService.IsAvailableSeatsValid(dto.AvailableSeats, dto.Id))
+        var workshop = await combinedWorkshopService.GetById(dto.Id, true).ConfigureAwait(false);
```

Comment:
Similar "design issue" as already mentioned. Btw, code duplication, you have here and in V1 controller, should trigger your attention that something is going wrong/misplaced/etc